### PR TITLE
docs: add a note for use functional component

### LINKED ISSUE #915: Defining functional components in router configuration will cause routing switching errors
### Version
4.0.6

### Reproduction link
[https://jsfiddle.net/starfishing/bcst21vg/](https://jsfiddle.net/starfishing/bcst21vg/)

### Steps to reproduce
change foor and bar

### What is expected?
router change is good

### What is actually happening?
vnode.component.isUnmounted has error

---
when a component like this:
```
function empty() {
// or return null
  return <div>hello</div>;
}
export default empty;
```

the router will change error

### PR DESCRIPTION
**Pull request by PeterAlfredLee**
Close #915

### CODE REVIEW

File: docs/api/index.md [hunk 1]
```diff
@@ -857,6 +857,17 @@ Route record that can be provided by the user when adding routes via the [`route
...
 - **See Also**: [Meta fields](../guide/advanced/meta.md)
 
+:::tip
+If you want to use a functional component, make sure to add a `displayName` to the component.
+
+For example:
+```js
+const Functional = () => h('div', 'functional')
+Functional.displayName = 'Functional'  // add a "displayName" to the component.
+const router = createRouter({ path: '/foo', component: Functional })
```
Comment (posva) [inline docs/api/index.md:867]: can you replace this with a `const routes = [{ ... }` like in the rest of the docs?

File: docs/api/index.md [hunk 2]
```diff
@@ -857,6 +857,17 @@ Route record that can be provided by the user when adding routes via the [`route
 
 - **See Also**: [Meta fields](../guide/advanced/meta.md)
 
+:::tip
+If you want to use a functional component, make sure to add a `displayName` to the component.
+
+For example:
+```js
+const Functional = () => h('div', 'functional')
```
Comment (posva) [inline docs/api/index.md:865]: Something More contextual is better
```suggestion
const HomeView = () => h('div', 'HomePage')
```

File: docs/api/index.md (hunk 1 above)
Comment (posva) [inline docs/api/index.md:867]: Also having `/` instead of `/foo` would be better

**Review by posva — CHANGES_REQUESTED** [review]
Thanks!

**Comment by PeterAlfredLee** [comment]
@posva Thank you for review this. Just pushed again. Please check it.

File: docs/zh/api/index.md [hunk 3]
```diff
@@ -851,6 +851,17 @@ stringifyQuery?: (
 
 - **更多的内容请看**：[Meta 字段](../guide/advanced/meta.md)
 
+:::tip 注意
+如果你想使用函数式组件, 请确保在组件上添加一个`displayName`。
```
Comment (edison1105) [inline docs/zh/api/index.md:855]:
```suggestion
如果你想使用函数式组件, 请确保在组件上添加一个 `displayName`。
```

File: docs/zh/api/index.md [hunk 4]
```diff
@@ -851,6 +851,17 @@ stringifyQuery?: (
...
+
+例如:
+```js
+const HomeView = () => h('div', 'HomePage')
+HomeView.displayName = 'HomeView'  // 在组件上添加一个`displayName`
```
Comment (edison1105) [inline docs/zh/api/index.md:860]:
```suggestion
HomeView.displayName = 'HomeView'  // 在组件上添加一个 `displayName`
```

**Review by edison1105 — CHANGES_REQUESTED** [review]

**Review by edison1105 — APPROVED** [review]

File: docs/api/index.md [hunk 5]
```diff
@@ -857,6 +857,17 @@ Route record that can be provided by the user when adding routes via the [`route
 
 - **See Also**: [Meta fields](../guide/advanced/meta.md)
 
+:::tip
+If you want to use a functional component, make sure to add a `displayName` to the component.
```
Comment (posva) [inline docs/api/index.md:861]:
```suggestion
If you want to use a functional component, make sure to add a `displayName` to it.
```

File: docs/api/index.md [hunk 6]
```diff
@@ -857,6 +857,17 @@ Route record that can be provided by the user when adding routes via the [`route
...
+
+For example:
+```js
+const HomeView = () => h('div', 'HomePage')
+HomeView.displayName = 'HomeView'  // add a "displayName" to the component.
```
Comment (posva) [inline docs/api/index.md:866]:
```suggestion
// in TypeScript, you will need to use the FunctionalComponent type
HomeView.displayName = 'HomeView'
```

**Review by posva — CHANGES_REQUESTED** [review]
Thanks! Can you adapt the last few changes on the Chinese doc as well, please?

File: docs/api/index.md (hunk 5 above)
Comment (PeterAlfredLee) [reply docs/api/index.md]: ok

File: docs/api/index.md (hunk 6 above)
Comment (PeterAlfredLee) [reply docs/api/index.md]: ok

**Review by posva — APPROVED** [review]
Thanks!

**Pull request merged**